float32 threshold
---
bool success
float32 new_threshold
